### ISSUE TITLE
Model training does not train well on custom dataset POSEC3D

### ISSUE THREAD

**Issue by mukeshnarendran7** [kennymckormick/pyskl#106]
I used the tutorial from mmaction docs POSEC3D. Aim is to classify lameness (i use two classes (0,1))
My config: I have annotated 2D keypoints that i used to develop the .pkl file according tommaction2 tutorial. Just adding reference config file and outcome. Would be great to hear what I am missing.
```
model = dict(
    type='Recognizer3D',
    backbone=dict(
        type='ResNet3dSlowOnly',
        depth=50,
        pretrained=None,
        in_channels=17,
        base_channels=32,
        num_stages=3,
        out_indices=(2, ),
        stage_blocks=(4, 6, 3),
        conv1_stride_s=1,
        pool1_stride_s=1,
        inflate=(0, 1, 1),
        spatial_strides=(2, 2, 2),
        temporal_strides=(1, 1, 2),
        dilations=(1, 1, 1)),
    cls_head=dict(
        type='I3DHead',
        in_channels=512,
        num_classes=2, # Change here
        spatial_type='avg',
        dropout_ratio=0.5),
    train_cfg=dict(),
    test_cfg=dict(average_clips='prob'))

dataset_type = 'PoseDataset'
ann_file_train = 'custom_dataset_train.pkl'
ann_file_val = 'custom_dataset_val.pkl'
left_kp = [1, 3, 5, 7, 9, 11, 13, 15]
right_kp = [2, 4, 6, 8, 10, 12, 14, 16]
train_pipeline = [
    dict(type='UniformSampleFrames', clip_len=48),
    dict(type='PoseDecode'),
    dict(type='PoseCompact', hw_ratio=1., allow_imgpad=True),
    dict(type='Resize', scale=(-1, 64)),
    dict(type='RandomResizedCrop', area_range=(0.56, 1.0)),
    dict(type='Resize', scale=(56, 56), keep_ratio=False),
    #dict(type='Flip', flip_ratio=0.5, left_kp=left_kp, right_kp=right_kp),
    dict(
        type='GeneratePoseTarget',
        sigma=0.6,
        use_score=True,
        with_kp=True,
        with_limb=False),
    dict(type='FormatShape', input_format='NCTHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
val_pipeline = [
    dict(type='UniformSampleFrames', clip_len=48, num_clips=1, test_mode=True),
    dict(type='PoseDecode'),
    dict(type='PoseCompact', hw_ratio=1., allow_imgpad=True),
    dict(type='Resize', scale=(-1, 64)),
    dict(type='CenterCrop', crop_size=64),
    dict(
        type='GeneratePoseTarget',
        sigma=0.6,
        use_score=True,
        with_kp=True,
        with_limb=False),
    dict(type='FormatShape', input_format='NCTHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
test_pipeline = [
    dict(
        type='UniformSampleFrames', clip_len=48, num_clips=10, test_mode=True),
    dict(type='PoseDecode'),
    dict(type='PoseCompact', hw_ratio=1., allow_imgpad=True),
    dict(type='Resize', scale=(-1, 64)),
    dict(type='CenterCrop', crop_size=64),
    dict(
        type='GeneratePoseTarget',
        sigma=0.6,
        use_score=True,
        with_kp=True,
        with_limb=False,
        double=True,
        left_kp=left_kp,
        right_kp=right_kp),
    dict(type='FormatShape', input_format='NCTHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
data = dict(
    videos_per_gpu=16,
    workers_per_gpu=2,
    test_dataloader=dict(videos_per_gpu=1),
    train=dict(
        type=dataset_type,
        ann_file=ann_file_train,
        data_prefix='',
        class_prob={i: 1 + int(i >= 60)
                    for i in range(120)},
        pipeline=train_pipeline),
    val=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        data_prefix='',
        pipeline=val_pipeline),
    test=dict(
        type=dataset_type,
        ann_file=ann_file_val,
        data_prefix='',
        pipeline=test_pipeline))
# optimizer
optimizer = dict(
    type='SGD', lr=0.0001, momentum=0.9,
    weight_decay=0.0003)  # this lr is used for 8 gpus
optimizer_config = dict(grad_clip=dict(max_norm=40, norm_type=2))
# learning policy
lr_config = dict(policy='CosineAnnealing', by_epoch=False, min_lr=0)
total_epochs = 70
checkpoint_config = dict(interval=10)
workflow = [('train', 10)]
evaluation = dict(
    interval=10,
    metrics=['top_k_accuracy', 'mean_class_accuracy'],
    topk=(1, 5))
log_config = dict(
    interval=20, hooks=[
        dict(type='TextLoggerHook'),
    ])
dist_params = dict(backend='nccl')
log_level = 'INFO'
work_dir = './work_dirs/posec3d/slowonly_r50_u48_240e_ntu120_xsub_keypoint'
load_from = 'slowonly_r50_u48_240e_ntu120_xsub_keypoint-6736b03f.pth'
resume_from = None
find_unused_parameters = False
```
```
2022-12-08 17:19:04,812 - mmaction - INFO - Evaluating top_k_accuracy ...
2022-12-08 17:19:04,812 - mmaction - INFO - 
top1_acc        0.5385
top5_acc        1.0000
2022-12-08 17:19:04,812 - mmaction - INFO - Evaluating mean_class_accuracy ...
2022-12-08 17:19:04,812 - mmaction - INFO - 
mean_acc        0.5000
2022-12-08 17:19:04,874 - mmaction - INFO - Now best checkpoint is saved as best_top1_acc_epoch_10.pth.
2022-12-08 17:19:04,874 - mmaction - INFO - Best top1_acc is 0.5385 at 10 epoch.
2022-12-08 17:19:04,874 - mmaction - INFO - Epoch(val) [10][1]  top1_acc: 0.5385, top5_acc: 1.0000, mean_class_accuracy: 0.5000
2022-12-08 17:20:12,297 - mmaction - INFO - Saving checkpoint at 20 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 13/13, 18.8 task/s, elapsed: 1s, ETA:     0s2022-12-08 17:20:13,110 - mmaction - INFO - Evaluating top_k_accuracy ...
2022-12-08 17:20:13,111 - mmaction - INFO - 
top1_acc        0.5385
top5_acc        1.0000
2022-12-08 17:20:13,111 - mmaction - INFO - Evaluating mean_class_accuracy ...
2022-12-08 17:20:13,111 - mmaction - INFO - 
mean_acc        0.5000
2022-12-08 17:20:13,111 - mmaction - INFO - Epoch(val) [20][1]  top1_acc: 0.5385, top5_acc: 1.0000, mean_class_accuracy: 0.5000
2022-12-08 17:21:20,248 - mmaction - INFO - Saving checkpoint at 30 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 13/13, 18.8 task/s, elapsed: 1s, ETA:     0s2022-12-08 17:21:21,056 - mmaction - INFO - Evaluating top_k_accuracy ...
2022-12-08 17:21:21,056 - mmaction - INFO - 
top1_acc        0.5385
top5_acc        1.0000
2022-12-08 17:21:21,056 - mmaction - INFO - Evaluating mean_class_accuracy ...
2022-12-08 17:21:21,057 - mmaction - INFO - 
mean_acc        0.5000
2022-12-08 17:21:21,057 - mmaction - INFO - Epoch(val) [30][1]  top1_acc: 0.5385, top5_acc: 1.0000, mean_class_accuracy: 0.5000
2022-12-08 17:22:29,102 - mmaction - INFO - Saving checkpoint at 40 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 13/13, 17.4 task/s, elapsed: 1s, ETA:     0s2022-12-08 17:22:29,965 - mmaction - INFO - Evaluating top_k_accuracy ...
2022-12-08 17:22:29,966 - mmaction - INFO - 
top1_acc        0.5385
top5_acc        1.0000
2022-12-08 17:22:29,966 - mmaction - INFO - Evaluating mean_class_accuracy ...
2022-12-08 17:22:29,966 - mmaction - INFO - 
mean_acc        0.5000
2022-12-08 17:22:29,966 - mmaction - INFO - Epoch(val) [40][1]  top1_acc: 0.5385, top5_acc: 1.0000, mean_class_accuracy: 0.5000
2022-12-08 17:23:38,353 - mmaction - INFO - Saving checkpoint at 50 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 13/13, 17.1 task/s, elapsed: 1s, ETA:     0s2022-12-08 17:23:39,238 - mmaction - INFO - Evaluating top_k_accuracy ...
2022-12-08 17:23:39,238 - mmaction - INFO - 
top1_acc        0.5385
top5_acc        1.0000
2022-12-08 17:23:39,238 - mmaction - INFO - Evaluating mean_class_accuracy ...
2022-12-08 17:23:39,238 - mmaction - INFO - 
mean_acc        0.5000
2022-12-08 17:23:39,239 - mmaction - INFO - Epoch(val) [50][1]  top1_acc: 0.5385, top5_acc: 1.0000, mean_class_accuracy: 0.5000
2022-12-08 17:24:47,316 - mmaction - INFO - Saving checkpoint at 60 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 13/13, 18.1 task/s, elapsed: 1s, ETA:     0s2022-12-08 17:24:48,164 - mmaction - INFO - Evaluating top_k_accuracy ...
2022-12-08 17:24:48,164 - mmaction - INFO - 
top1_acc        0.5385
top5_acc        1.0000
2022-12-08 17:24:48,165 - mmaction - INFO - Evaluating mean_class_accuracy ...
2022-12-08 17:24:48,165 - mmaction - INFO - 
mean_acc        0.5000
2022-12-08 17:24:48,165 - mmaction - INFO - Epoch(val) [60][1]  top1_acc: 0.5385, top5_acc: 1.0000, mean_class_accuracy: 0.5000
2022-12-08 17:25:55,887 - mmaction - INFO - Saving checkpoint at 70 epochs
[>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>] 13/13, 18.2 task/s, elapsed: 1s, ETA:     0s2022-12-08 17:25:56,747 - mmaction - INFO - Evaluating top_k_accuracy ...
2022-12-08 17:25:56,747 - mmaction - INFO - 
top1_acc        0.5385
top5_acc        1.0000
2022-12-08 17:25:56,747 - mmaction - INFO - Evaluating mean_class_accuracy ...
2022-12-08 17:25:56,747 - mmaction - INFO - 
mean_acc        0.5000
2022-12-08 17:25:56,748 - mmaction - INFO - Epoch(val) [70][1]  top1_acc: 0.5385, top5_acc: 1.0000, mean_class_accuracy: 0.5000
```
Also get this class prob as 1 and 2. Not sure what it means. 
```
data = dict(
    videos_per_gpu=16,
    workers_per_gpu=2,
    test_dataloader=dict(videos_per_gpu=1),
    train=dict(
        type='PoseDataset',
        ann_file=
        'local path/custom_dataset_train.pkl',
        data_prefix='',
        class_prob=dict({
            0: 1,
            1: 1,
            60: 2,
            61: 2,
```

**Comment by kennymckormick** [comment]
Hi, [mukeshnarendran7](https://github.com/mukeshnarendran7), without enough context I'm not sure what is the problem, but a reasonable guess is that you have too little training examples, and the learning rate is too small (you set it as 1e-4). It seems that the model didn't learn anything (validation acc is always the same).

**Comment by mukeshnarendran7** [comment]
Hey [kennymckormick](https://github.com/kennymckormick) thanks for getting back. I for sure have very little training and test sample and I guess it might relate to the low validation loss. 
Is there a RGB-POSEC3D model for usage or the models in this repo (posec3d) pertain only to skeletal action recognition via joint info (x,y, z)

**Comment by kennymckormick** [comment]
Now we also provide RGBPoseConv3D in PYSKL.
